src/main/java/org/jabref/gui/fieldeditors/IdentifierEditorViewModel.java

```diff
@@ -78,6 +94,18 @@ public void openExternalLink() {
         );
     }
 
+    public void openDOIWithCustomBase(String baseURI) {
+        identifier.get().map(x -> (DOI) x).flatMap(x -> x.getExternalURIWithCustomBase(baseURI)).ifPresent(
```

Comment:
```suggestion
        identifier.get().map(identifier-> (DOI) identifier).flatMap(DOI::getExternalURIWithCustomBase(baseURI)).ifPresent(
```

please no single char variable name like x, use a meaninfgul variable and inside the flatmap you should be able to use a static syntax like DOI::getExternalUriWithCustomBase

Reply:
Correct me if I'm wrong but since flatMap requires Function and we have two input arguments (String baseUri and DOI) is it not impossible to implement the static syntax? IntelliJ seems to agree with that.

Reply:
Sorry, overlooked that, you are right